tests/moses/scoring/behave_bg_cscoreUTest.cxxtest

```diff
@@ -0,0 +1,162 @@
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <cxxtest/TestSuite.h>
+#include <opencog/asmoses/moses/scoring/behave_bg_cscore.h>
+#include <iostream>
+#include <opencog/util/Logger.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <opencog/guile/SchemeEval.h>
+#include <opencog/asmoses/combo/combo/combo.h>
+#include <opencog/asmoses/combo/combo/iostream_combo.h>
+#include <opencog/atoms/atom_types/atom_types.h>
+#include <opencog/asmoses/data/table/table.h>
+#include <opencog/asmoses/data/table/table_io.h>
+#include <opencog/asmoses/moses/scoring/bscores.h>
+#include <string>
+#include <opencog/asmoses/moses/moses/background_feature.h>
+#include <opencog/asmoses/moses/main/populate_atomspace.h>
+
+using namespace opencog;
+using namespace std;
+using namespace opencog::combo;
+using namespace opencog::atomese;
+using namespace opencog::moses;
+
+typedef compressed_truth_table_bscore table_bscore;
+
+const string utest_dir = string(PROJECT_SOURCE_DIR) + "/tests/moses/scoring/";
+const string target_feature = "out";
+
+class BehaveBGCscoreUTest : public CxxTest::TestSuite
+{
+private:
+	AtomSpace _as;
+public:
+	BehaveBGCscoreUTest() {
+
+	}
+
+	~BehaveBGCscoreUTest() {
+		if (!CxxTest::TestTracker::tracker().suiteFailed())
+			remove(logger().get_filename().c_str());
+
+	}
+
+	void setUp();
+
+	void tearDown();
+
+	void load_scm_file(const string& path, SchemeEval& eval);
+
+	void test_atomese_program();
+
+	void test_combo_program();
+
+
+};
+
+void BehaveBGCscoreUTest::setUp()
+{
+	_as.clear();
+}
+
+void BehaveBGCscoreUTest::tearDown() {}
+
+void BehaveBGCscoreUTest::load_scm_file(const string &path, SchemeEval& eval)
+{
+	eval.eval("");
+	eval.eval(string("(add-to-load-path \"") + utest_dir + string("\")"));
+	string rv = eval.eval("(use-modules (opencog))");
+	TS_ASSERT(not eval.eval_error());
+
+	rv = eval.eval(string("(load-from-path \"") + path + string("\")"));
+	cout << "Result from loading file is %s\n" << rv << endl;
+    TS_ASSERT(not eval.eval_error());
+}
+
+void BehaveBGCscoreUTest::test_atomese_program()
+{
+    logger().set_print_to_stdout_flag(true);
+	Table tt = loadTable(utest_dir + "data_bg.csv", target_feature);
+	CompressedTable ctt = tt.compressed();
+	table_bscore ct_bscore(ctt);
+	Types tts = {ATTRACTION_LINK};
+	vector<string> labels = {"f1", "f2", "f3", "f4"};
+	SchemeEval eval(&_as);
+	load_scm_file("data.scm", eval);
+    behave_bg_cscore bscore(ct_bscore, &_as, CONCEPT_NODE, tts, labels);
```

Comment:
Indentation.

@Habush, you probably want to configure you editor/IDE to explicitly show tabs, that way you'd be able to spot inconsistency with your naked eye.